Sources/Decodable.swift

```diff
@@ -36,12 +36,34 @@ extension Dictionary where Key: Decodable, Value: Decodable {
     }
 }
 
+public typealias DecodeArrayInvalidObjectReporter = (AnyObject, ErrorType) -> Void
```

Comment:
I don't think the purpose of the callback is of importance to the method, whether it's to do logging or anything else, so I made it purposely vague. It could be `Callback` instead of `Reporter`, but I liked the verb `report` to mean "tell the caller about something it found".